pkgs/by-name/hy/hyprwall/package.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  fetchurl,
+  fetchFromGitHub,
+  rustPlatform,
+  makeDesktopItem,
+  copyDesktopItems,
+
+  pkg-config,
+  glib,
+  gdk-pixbuf,
+  pango,
+  graphene,
+  gtk4,
+  gtk3,
+  git,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "hyprwall";
+  version = "0.1.8";
+
+  src = fetchFromGitHub {
+    owner = "hyprutils";
+    repo = "hyprwall";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-dzPd+5cws3hKhdd1CKKEO7EWMS0XW0y1vqxg1XKX+gY=";
+  };
+
+  cargoHash = "sha256-gT2ysWHckcUl1yx5tciy6kSvZZ0srrs4OwI1mr/58Pc=";
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "HyprWall";
+      exec = "hyprwall";
+      icon = fetchurl {
+        url = "https://github.com/hyprutils/hyprwall/blob/main/hyprwall.png";
+        hash = "sha256-YkggQGJiN3TaBTW8aaorBzl5pJwf9fiU0ZnLCj6knhk=";
+      };
+      type = "Application";
+      terminal = false;
+      comment = "GUI for setting wallpapers with Hyprpaper, written in blazingly fast Rust!";
+      desktopName = "HyprWall";
+      categories = [
+        "Utility"
+      ];
+      keywords = [
+        "wallpaper"
+        "hyprwall"
+      ];
+    })
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    glib
+    gdk-pixbuf
+    pango
+    copyDesktopItems
+    graphene
+    gtk4
+    gtk3
+    git
+  ];
+
+  buildInputs = [
+    pkg-config
+    glib
+    gdk-pixbuf
+    pango
+    graphene
+    gtk4
+    gtk3
+  ];
+
+  meta = {
+    description = "GUI for setting wallpapers with hyprpaper, swww, swaybg, wallutils, and feh. Written in blazingly fast Rust";
```

Comment:
`meta.description` must "Be short, just one sentence"

https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes